resource/select_gpu_devices.py

```diff
@@ -0,0 +1,35 @@
+# -*- coding: utf-8 -*-
+import re
+import dataiku
+from constants import GPU_CONFIGURATION
+from safe_logger import SafeLogger
+
+
+logger = SafeLogger("Forecast plugin")
+
+
+def do(payload, config, plugin_config, inputs):
+    """ Retrieve a list of available GPU devices. """
+    choices = []
+
+    if payload.get("parameterName") == "gpu_devices":
+        try:
+            import mxnet as mx
+        except OSError as cuda_error:  # error when importing mxnet
+            logger.error(f"Error when importing mxnet: {cuda_error}")
+            choices += [{"label": "No GPU detected on DSS server, please check your server CUDA installation", "value": GPU_CONFIGURATION.NO_GPU}]
+        else:
+            try:
+                num_gpu = mx.context.num_gpus()
+            except mx.base.MXNetError as num_gpus_error:  # error on num_gpus()
+                logger.error(f"Error when querying number of GPU: {num_gpus_error}")
+                choices += [{"label": "No GPU detected on DSS server, please check your server CUDA installation", "value": GPU_CONFIGURATION.NO_GPU}]
+            else:
+                if num_gpu > 0:
+                    choices += [{"label": f"GPU #{n}", "value": n} for n in range(num_gpu)]
+                else:
+                    choices += [{"label": "No GPU detected on DSS server, please check that your server has GPUs", "value": GPU_CONFIGURATION.NO_GPU}]
+        finally:
+            choices += [{"label": "Select a GPU from the container configuration", "value": GPU_CONFIGURATION.CONTAINER_GPU}]
```

Comment:
After rethinking about this again, I don't like the idea of always adding this option to the multiselect. With this setup, this option would always appear, despite most customers not having GPU-enabled container configuration. By satisfying a minority, we would confuse the majority.
I would rather add a boolean "Use GPU from container" / description: "Experimental - Requires a GPU-enabled container configuration"
Activating this boolean would make the multiselect invisible